ethereum/contracts/IncentivizedInboundChannel.sol

```diff
@@ -74,8 +91,15 @@ contract IncentivizedInboundChannel is InboundChannel {
             (bool success, ) =
                 _messages[i].target.call{value: 0, gas: MAX_GAS_PER_MESSAGE}(_messages[i].payload);
 
+            _rewardAmount = _rewardAmount + _messages[i].fee;
             emit MessageDispatched(_messages[i].nonce, success);
         }
+
+        // Attempt to reward the relayer
+        try rewardSource.reward(_relayer, _rewardAmount) { }
+        catch {
+            emit RelayerNotRewarded(_relayer, _rewardAmount);
+        }
```

Comment:
Relayers may not want others to know that they are being rewarded certain sums though? I think we discussed that when I implemented the other direction (eth->sub fees/rewards). I can add an event if you think otherwise though.

Reply:
It is transparent anyways, e. g. through call tracing. 

I don't really see a strong use case for having an event, and since it costs gas, it would be fine to leave it as is.

Reply:
when would this case ever occur? if it does we have a serious problem and should basically pause/shut down the bridge to allow for recovery from governance?

Reply:
> when would this case ever occur? if it does we have a serious problem and should basically pause/shut down the bridge to allow for recovery from governance?

The `RelayerNotRewarded` event is just so I can verify the unhappy path in unit tests. It may go away in future.

Agree that we should somehow shutdown the bridge when we hit this unhappy path, but I think that's better to implement in another PR.

Reply:
@vgeddes otherwise we can just remove the try catch as this should never happen?

Reply:
can also remove that test, as its not really an unhappy path test, it's testing something that should never happen